module/Finna/src/Finna/RecordDriver/SolrLido.php

```diff
@@ -2055,38 +2093,96 @@ protected function getLanguageSpecificItem($element, $language)
+            ->objectIdentificationWrap->objectDescriptionWrap->objectDescriptionSet
+            ?? [] as $node) {
+            $resultsUntyped[] = $node->descriptiveNoteValue;
+            if ($node->attributes()->type == 'description') {
+                $desc1[] = $node->descriptiveNoteValue;
+            }
+            if ($node->attributes()->type == null) {
+                $desc2[] = $node->descriptiveNoteValue;
+            }
+        }
+        foreach ($this->getXmlRecord()->lido->descriptiveMetadata->objectRelationWrap
+            ->subjectWrap->subjectSet ?? [] as $node) {
+            if ($node->displaySubject) {
+                if ($node->displaySubject->attributes()->label == 'aihe') {
+                    $subjct[] = $node->displaySubject;
+                    $resultsUntyped[] = $node->displaySubject;
+                }
+                if ($node->displaySubject->attributes()->label == '') {
+                    $resultsUntyped[] = $node->displaySubject;
                 }
             }
         }
-
-        $preferredLanguages = $this->getPreferredLanguageCodes();
-        foreach ($this->getXmlRecord()->xpath(
-            'lido/descriptiveMetadata/objectIdentificationWrap/objectDescriptionWrap'
-            . '/objectDescriptionSet[@type="description"]/descriptiveNoteValue'
-        ) as $node) {
-            if (in_array((string)$node->attributes()->lang, $preferredLanguages)) {
-                if ($term = trim((string)$node)) {
-                    $results[] = $term;
+        // Set results according to found items and their set order
+        if ($desc1 || $desc2 || $subjct) {
+            $terms = $this->getAllLanguageSpecificItems(
+                $desc1 ?: $desc2 ?: $subjct,
+                $language
+            );
+            if (!$subjct) {
+                foreach ($terms as $item) {
+                    foreach ($item as $part) {
+                        $results[] = (string)$part;
+                    }
+                }
+            }
+            if ($subjct) {
+                $titleToCheck = $this->getLanguageSpecificItem($subjct, $language);
+                $checkTitle = str_replace([',', ';'], ' ', (string)$titleToCheck)
+                    != $title;
+                if ($terms && $checkTitle) {
+                    foreach ($terms as $item) {
+                        foreach ($item as $part) {
+                            $results[] = (string)$part;
+                        }
+                    }
                 }
             }
         }
-
-        if (!$results && !empty($this->fields['description'])) {
-            $results[] = (string)($this->fields['description']) != $title
-                ? (string)$this->fields['description'] : '';
+        // If no specified fields found, set results as description-field content
+        if ($resultsUntyped && !$results) {
+            $checkDesc = [];
+            foreach ($resultsUntyped as $item) {
```

Comment:
Mitä tässä tapahtuu? Mitä `$checkDesc` ja `$replaceDesc` edustavat?

Reply:
String muodossa oleva resultsUntyped, jolla vertaan description kentän sisältöön. Toinen alkuperäisessä muodossa (checkdesc) ja toinen kieliversiollinen (replaceDesc), jolla se korvataan?

Reply:
(Näin yritin ratkaista sitä ongelmaa, jos description kentässä (string) on jo se splitTitlen osa, mutta pitäisi saada sen lisäksi vain kieliversiot.)